website/cue/reference/components.cue

```diff
@@ -178,6 +178,8 @@ components: {
 		}
 		let Args = _args
 
+		auto_generated: bool | *false
```

Comment:
We use this new field to annotate sources/sinks as being auto-generated.

Essentially, a lot of the configuration fields for components are automatically added if the right "feature" field is enabled, or configured specifically, in the per-component Cue file. Additionally, different sections of the page (like "How It Works") are only added if the feature is correspondingly enabled.

In order to avoid having to rework all of that stuff, and leave un-converted components without relevant fields during the migration process, this new field was added which defaults to `false`. Sources and sinks now add these feature-specific fields if `auto_generated` is `false`, but omits them when `true`. For things like the `How It Works` section, though, it always emits those even for sources/sinks where `auto_generated` is `true`.

Ultimately, this should allow for the minimum number of visible changes to sources/sinks which have not yet been converted to use the machine-generated base Cue documentation.